ext4/internal/gpt/types.go

```diff
@@ -0,0 +1,97 @@
+	StartingLBA   uint32  // 4 bytes, set to 0x00000001 (LBA of the GPT partition header)
+	SizeInLBA     uint32  // 4 bytes, set to the size of the disk - 1 or 0xffffffff if size is too big to represent here
+}
+
+// GPT info: GPT primary table must be in LBA 1 (aka the second logical block) and the
+// secondary (alternate) table must be in the last LBA of the disk
+type Header struct {
+	Signature                uint64    // 8 bytes, ASCII string "EFI PART"
+	Revision                 uint32    // 4 bytes, 0x00010000
+	HeaderSize               uint32    // 4 bytes, must be greater than or equal to 92 and must be less than or equal to the logical block size
+	HeaderCRC32              uint32    // 4 bytes, CRC32 checksum for the GPT header structure. Computed by setting this to zer0, and computing the 32 bit crc for headersize in bytes
+	ReservedMiddle           uint32    // 4 bytes, must be zzero
+	MyLBA                    uint64    // 8 bytes, The LBA that contains this data structure
+	AlternateLBA             uint64    // 8 bytes, LBA of the alternate GPT header
+	FirstUsableLBA           uint64    // 8 bytes, the first logical block that may be used by a GPT entry, must be >=34
+	LastUsableLBA            uint64    // 8 bytes, the last usable logical block to be used by a GPT entry
+	DiskGUID                 guid.GUID // 16 bytes, used to uniquely identify the disk
+	PartitionEntryLBA        uint64    // 8 bytes, the starting LBA of the GPT Entries Array
+	NumberOfPartitionEntries uint32    // 4 bytes, the number of partition entries
+	SizeOfPartitionEntry     uint32    // 4 bytes, the size in bytes of each of the partition entry structures in the Entry Array. Must be set to a value of 128 x 2^n, where n is >= 0
+	PartitionEntryArrayCRC32 uint32    // 4 bytes, the crc32 of the entry array. Starts at PartitionEntryLBA and is computed over a byte length of NumberOfPartitionEntries * SizeOfPartitionEntry
+	ReservedEnd              [420]byte // rest of the block, BlockSize in bytes - 92, must be set to 0
+}
+
+type PartitionEntry struct {
+	PartitionTypeGUID   guid.GUID // 16 bytes, unique ID that defines the purpose and type of this partition.
+	UniquePartitionGUID guid.GUID // 16 bytes, unique for every partition entry, must be assigned when the entry is created
+	StartingLBA         uint64    // 8 bytes, Starting LBA of the parition defined by this entry
+	EndingLBA           uint64    // 8 bytes, Ending LBA of the partition
+	Attributes          uint64    // 8 bytes, attribute bits
+	PartitionName       [72]byte  // 72 bytes, null terminated string with the name
+}
+
+// The layout of a GPT disk is as follows:
+// | Protective MBR 				| - 1 block
+// | Partition Table HDR 			| - 1 block
+// | Partition Entry Array 			| - Size of Partition Entry * number of partitions
+// | Partition 0 					|
+// | ...... 						|
+// | Partition n 					|
+// | Backup Partition Entry Array 	|
+// | Backup Partition Table HDR 	| - Last 1 block
```

Comment:
Is github rendering the tabs weird, or do the ends not line up?

Reply:
They show up correctly in my vscode :(

Reply:
Don't use tabs for alignment of text, use spaces instead :)